### ISSUE TITLE
APhotoManager: photo not found when opened from file manager with app specific content uri-s

### ISSUE THREAD

**Issue by kodeblacc** [k3b/APhotoManager#118]
#### Expected behavior
I expected I'll be able to view pictures from dir filr manager using Aphotomanager

#### Actual behavior
When I choose to view a picture from dir file manager with Aphotomanager it's say photo not found in media database and opens another photo

#### How to reproduce
- download Aphotomanager and Dir file manager both on Fdroid.
- access a directory on dir filemanager with more than 10 pictures
- select the option to open picture with A photoviewer
- it says photo not found in media database.
<hr>

* Device: Infinix Note 4
* Operating system: Android 8.0
*Software version: Aphoto manager 0.6.4.180314
Dir File manager 1.5.5

#### Recording Of The Bug
https://youtu.be/WAPylthmLbc
    

<br /><hr/><em>Posted on <a href="https://utopian.io/utopian-io/@kodeblacc/aphoto-manager-says-photo-not-found-in-media-database-when-opened-from-dir-file-manager">Utopian.io -  Rewarding Open Source Contributors</a></em><hr/>

-----
[update k3b 2018-05-03]

Reproducable: OI-Filemanger tries to open content://org.openintents.filemanager//storage/emulated/0/DCIM/..... which APhotoManager currently does not understand.

**Comment by k3b** [comment]
Thanks for the report.

Can you help me to find the cause:

* Open "A Photo Manger" 
* goto settings menu 
* select all checkboxes under "Diagnose: LogCatXXXX"
* press "Delete Logcat"
* exit settings
* exit "A Photo Manger" 
* repeat the actions that causes the error
* Open "A Photo Manger" again
* goto settings menu 
* press "save logcat"
* open a filemanager at directory [ExternalStorageDirectory]/copy/log/
* there should be one ore more files androFotofinder.logcat-2018....txt

These files may contain the infos what went wrong

Can you attach the newest two files to this ticket.

Then i can analyse what went wrong

**Comment by kodeblacc** [comment]
Would have done that, unfortunately opening settings crashes the app

**Comment by k3b** [comment]
Sorry. Version "Aphoto manager 0.6.4.180314" had a bug in the settings #116 .

On f-droid and up2down there is a fixed "version 0.6.4.180321"

**Comment by kodeblacc** [comment]
It works now

**Comment by kodeblacc** [comment]
Can't see delete logcat

**Comment by k3b** [comment]
in english the menu item is called "clear logcat".

If the problem is gone away. i will close this ticket for now.

feel free to reopen it, if the issue happens again.

thanks for cooperation.

**Comment by kodeblacc** [comment]
This issue is still happening, even in the latest release of the software and logcat doesn't reveal the problem.

It is reproducible with Dir File Manager and OI file manager

**Comment by kodeblacc** [comment]
This is the logcat report I got and it doesn't contain useful information about what's causing this issue.

--------- beginning of main
[ 04-19 06:23:46.638  8583: 8583 E/k3bFoto  ]
SettingsActivity-SaveLogCat()

[ 04-19 06:23:46.639  8583: 8583 E/k3bFoto  ]
saving errorlog ('LocCat') to /storage/emulated/0/copy/log/androFotofinder.logcat-20180419-052346.txt

[ 04-19 06:24:27.710  8583: 8583 E/k3bFoto  ]
SettingsActivity-SaveLogCat()

[ 04-19 06:24:27.710  8583: 8583 E/k3bFoto  ]
saving errorlog ('LocCat') to /storage/emulated/0/copy/log/androFotofinder.logcat-20180419-052427.txt

[ 04-19 06:25:57.933  8583: 8583 E/k3bFoto  ]
SettingsActivity-SaveLogCat()

[ 04-19 06:25:57.934  8583: 8583 E/k3bFoto  ]
saving errorlog ('LocCat') to /storage/emulated/0/copy/log/androFotofinder.logcat-20180419-052557.txt

**Comment by k3b** [comment]
thanks for reporting. Under [settings](https://github.com/k3b/APhotoManager/wiki/settings): below Diagnostics: have you enabled all checkboxes "LogCat XXXX" before reproducing the problem? 

* goto settings:
* Clear LogCat
* enabled all checkboxes "LogCat XXXX" 
* leave settings
* reproduce the problem
* goto settings 
* Save LogCat to file

this should record what is happening.

**Comment by k3b** [comment]
i have looked at the video it shows "photo not found in the media database //storage/emulated/0/DCIM/IMG.....jpg"

The file url should be "/storage/emulated/0/DCIM" and not "//storage/emulated/0/DCIM". Which filemanager are you using?

Can you open the image from the filemanager with other gallery apps?

**Comment by k3b** [comment]
From https://steemit.com/utopian-io/@sammaz/when-i-tried-to-open-image-in-oi-file-manager-through-a-photo-manager-it-is-not-showing-images-properly

Here the problem is reproducable because oi-file-manager uses INTENT-VIEW with url 
content://org.openintents.filemanager/%2fstorage%2femulated%2f0%2fDCIM%2f..... which is converted to content://org.openintents.filemanager//storage/emulated/0/DCIM/.....
which becomes "//storage/emulated/0/DCIM/....."

a-photo-manager currently cannot translate oi-file-manager specific content url back to file url and therefore cannot open the jpg as a file.

Since new android versions do not like file uri-s "file:/path/to/file.jpg" modern android apps have to app specific content uri-s instead.

I will add a workaround that tries to get the original file uri from content://something//....

**Comment by kodeblacc** [comment]
Oh okay. It is reproducible with some other file manager too

**Comment by k3b** [comment]
I use an old filemanager that uses "file:/..." urls therefore it is working with it.

With the logcat as described my answer from 2018-04-21 i would have seen the problem immediately.

**Comment by k3b** [comment]
@kodeblacc hello kodeblacc feel free to go on. you should have write permission in the project-s wiki. (only for the wiki not for the sourcecode branch with the main readme.md) if this is not working please let me know.

Note: this is a hobby project with no budget so i cannot compensate you for your work.

---

I have looked at a privacy.md you already contributed for an other project which is exactly what this project is missing.

For the privacy statement: if settings ShowInMap/UseMapsforgeOfflineMap is **disabled** and some "map display" is shown the app may fetch mapdata from one of the openstreetmap servers for the current visible map area if this mapdata is currently not cached yet.

In other words: openstreetmap servers may contain logentries with your ip-adress and  date/time and geographic coordinate you where watching on the map. Each mapdata is only loaded once and then locally stored (a.k.a. data caching).

----

if you want to change something in the sourcecodebase outside the wiki (i.e. readme.md) feel free to clone and make a mergerequest.

**Comment by kodeblacc** [comment]
Thanks for replying. Like I said before, I'm not much of a developer so I'm
quite confused about the "project-s wiki" stuff. But what I can understand
is that you need a privacy policy document and an updated readme is not
needed. Am I correct?

Well I'll go ahead and create a pull request for a privacy document. Also
don't mind about paying me.
Thanks.
